Add support for binary and hex files

simonbothen-radinn opened the pull request:
In some cases it is more convinient to use a binary or hex file directly instead of using a zip file.

We can check the extention and do the appropriate action.

manualexSP commented:
I will have a look and do some tests in the next period. Thank you for contributing

mmesarosiuSP commented on android/src/main/java/com/pilloxa/dfu/RNNordicDfuModule.java:
```diff
@@ -38,7 +38,13 @@ public void startDFU(String address, String name, String filePath, Promise promi
         starter.setUnsafeExperimentalButtonlessServiceInSecureDfuEnabled(true);
-        starter.setZip(filePath);
+
+        if (filePath.endsWith(".bin") || filePath.endsWith(".hex")) {
```
On iOS the extension check is case insensitive, but on Android it isn't.  Wouldn't it be better to check against the filePath's toLowerCase()?

mmesarosiuSP commented:
I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12  updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2).
Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

simonbothen-radinn commented:
> I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12 updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2). Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

Thanks for the feedback. Will look into the above next week.

simonbothen-radinn commented:
> I've added your [repo's branch](https://github.com/simonbothen-radinn/react-native-nordic-dfu/tree/master_binary_support) as npm dependency, however on my Android 12 device it was crashing with your branch. I'm assuming because you don't have the Android 12 updates from this [PR](https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/2). Instead I applied these changes on our repo and installed it as a local npm package and it was working properly.

Rebased to remotes HEAD.

manualexSP commented:
Moved your implementation on this MR https://github.com/Salt-PepperEngineering/react-native-nordic-dfu/pull/6
